### PR TITLE
Install and configure Postgres

### LINKED ISSUE #2: install & configure postgres
specify tasks for installing postgres (& postgis) in `playbook.yaml`

### PR DESCRIPTION
**Pull request by TristanDamron**
Resolves #2 

Installs and configures Postgres and PostGIS for use in a geoCML deployment. This changeset installs the following software to the image:
- postgresql
- postgis (the geodata extension for Postgres)
- pgadmin4 

Additionally, this changeset adopts the community.postgresql Ansible package as a dependency allowing us to configure Postgres automatically after deployment.

Changes with this PR automatically create a Postgres database (geocml_db) and a database owner user (geocml) that users can use in production. There is also code that configures the default workspace user to switch to the postgres db admin user (postgres) without superuser privileges.

### CODE REVIEW

**Comment by TristanDamron** [comment]
@capsulecorplab I'm running into some limitations with Kasm atm... I can't seem to figure out how to start the postgresql service when the workspace boots. I did find some documentation about starting programs at boot time (https://kasmweb.com/docs/latest/how_to/building_images.html?highlight=custom_startup), however I need root privs to start the service which is not something that I can do using this feature as it's described in the documentation.

I'm not totally sure what the solution would be here--it's against Kasm's design philosophy to allow users to sudo, since that creates potential instability/mutations in the system. In a perfect world, I would like to see an option to run the custom_startup.sh script run with root privs and then boot into the workspace with a non-root user. That's not something directly in our control, though we could suggest it to the Kasm development team. 

Regardless of what we end up doing for this, I think that we'll need some good documentation :sweat_smile:

**Comment by TristanDamron** [comment]
@capsulecorplab This PR is finally ready for you to review! I've done some preliminary testing and it looks like everything is working okay. We should probably create a spin off issue for some documentation related to this PR as well (dependent on getting a documentation toolchain/doc-driven-development pattern set up). I'll leave that to you as you're the documentation expert here :smile: 

Let me know if this PR needs any cleaning up or if you run into major issues getting it working.

File: Dockerfile [hunk 1]
```diff
@@ -34,8 +35,8 @@ RUN rm -rf $HOME/install_files/
 RUN chown 1000:0 $HOME
 RUN $STARTUPDIR/set_user_permission.sh $HOME
 
-ENV HOME /home/kasm-user
+ENV HOME /home/kasm-user
```
Comment (capsulecorplab) [inline Dockerfile:38]: Trailing whitespace

File: Dockerfile [hunk 2]
```diff
@@ -34,8 +35,8 @@ RUN rm -rf $HOME/install_files/
...
 RUN $STARTUPDIR/set_user_permission.sh $HOME
 
-ENV HOME /home/kasm-user
+ENV HOME /home/kasm-user 
 WORKDIR $HOME
-RUN mkdir -p $HOME && chown -R 1000:0 $HOME
-
-USER 1000
\ No newline at end of file
+RUN mkdir -p $HOME && chown -R default $HOME
```
Comment (capsulecorplab) [inline Dockerfile:40]: Trailing whitespace

Comment (TristanDamron) [reply Dockerfile]: Fixed with [f075291](https://github.com/geoCML/kasm-gis-workspace/pull/6/commits/f0752918f5b3ec3e7cd23b807dfa74874d577d94)

File: Dockerfile (hunk 1 above)
Comment (TristanDamron) [reply Dockerfile]: [f075291](https://github.com/geoCML/kasm-gis-workspace/pull/6/commits/f0752918f5b3ec3e7cd23b807dfa74874d577d94)

**Comment by TristanDamron** [comment]
@capsulecorplab I've fixed the error you were running into with binding the postgresql service to the IPV6 localhost! I've also updated the README to include the new dependencies included with this PR.

**Comment by capsulecorplab** [comment]
Pulled image https://github.com/orgs/geoCML/packages/container/kasm-gis-workspace/32927596?tag=sha-68ecfda, ran `su - postgres` and started postgres with `service postgresql start`, still getting the binding error with IPv6
Screenshot from 2022-08-06 00-07-31

Service appears to be online with 
Screenshot from 2022-08-06 00-07-35

**Comment by TristanDamron** [comment]
It could be a cache issue. Could you run docker system prune -a and try again?

**Comment by capsulecorplab** [comment]
I think I resolved the issue! I needed to reset the persistence layer, i.e., `/kasm_user`. It seems to be "okay" now
Screenshot from 2022-08-06 10-45-58

**Comment by capsulecorplab** [comment]
Is there a postgis example I can run, just to see a simple use case?

**Comment by TristanDamron** [comment]
This video does a good job of explaining how to import a Shapefile (a kind of vector data) to a Postgres/PostGIS database: https://youtu.be/D2Z2udAKNeY

UCLA has a large amount of Shapefiles you can download to test with: https://apps.gis.ucla.edu/geodata/dataset?groups=global

Also, just as a reminder--to connect to the database in QGIS, you'll need to provide the host (127.0.0.1), the database name (geocml_db), the database owner username (geocml), and DBO password (geocml).

**Review by capsulecorplab — APPROVED** [review]
LGTM!

**Pull request merged**